src/mempool/mempool.rs

```diff
@@ -34,7 +37,52 @@ impl Default for Config {
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub struct MempoolKey {
-    inserted_at: Instant,
+    timestamp: u64,
+    identity: String,
+}
+
+impl MempoolKey {
+    pub fn new(timestamp: u64, identity: String) -> Self {
+        MempoolKey {
+            timestamp,
+            identity,
+        }
+    }
+}
+
+impl proto::Message {
+    pub fn mempool_key(&self) -> MempoolKey {
+        if let Some(data) = &self.data {
+            return MempoolKey::new(data.timestamp as u64, self.hex_hash());
```

Comment:
I think we should use the arrival timestamp and not the timestamp on the message itself for ordering. If we use the timestamp on the message, backdated messages will always be prioritized.

Reply:
The existing implementation is FIFO and so I was trying to preserve that behaviour, will leave a comment.

Reply:
https://github.com/farcasterxyz/snapchain/pull/200/commits/118c035be8bc035f2d262a5cbb23553651a3e8dc